source/docs/software/wpilib-tools/smartdashboard/test-mode-and-live-window/pid-tuning-with-smartdashboard.rst

```diff
@@ -0,0 +1,35 @@
+
+Finding the setpoint values with LiveWindow
+-------------------------------------------
+.. image:: images/PID-tuning-with-SmartDashboard/finding-setpoint-values.png
+
+Create a PID Subsystem for each mechanism with feedback. The PID Subsystems contain the actuator (motor) and the feedback sensor (potentiometer in this case). You can use Test mode to display the subsystem sensors and actuators. Using the slider manually adjust the actuator to each desired position. Note the sensor values (2) for each of the desired positions. These will become the setpoints for the PID controller.
+
+Viewing the PIDController in LiveWindow
+---------------------------------------
+.. image:: images/PID-tuning-with-SmartDashboard/viewing-pid-controller.png
+
+In Test mode, the PID Subsystems display their P, I, and D parameters that are set in the code. The P, I, and D values are the weights applied to the computed error (P), sum of errors over time (I), and the rate of change of errors (D). Each of those terms is multiplied by the weights and added together to form the motor value. Choosing the optimal P, I, and D values can be difficult and requires some amount of experimentation. The Test mode on the robot allows the values to be modified, and the mechanism response observed.
+
+Tuning the PIDController
+------------------------
+.. image:: images/PID-tuning-with-SmartDashboard/tuning-pid-controller.png
+
+Tuning the PID controller can be difficult and there are many articles that describe techniques that can be used. It is best to start with the P value first. To try different values fill in a low number for P, enter a setpoint determined earlier in this document, and note how fast the mechanism responds. If it responds too slowly, perhaps never reaching the setpoint, increase P. If it responds too quickly, perhaps oscillating, reduce the P value. Repeat this process until you get a response that is as fast as possible without oscillation. It's possible that having a P term is all that's needed to achieve adequate control of your mechanism.
+
+Once you have determined P, I, and D values they can be inserted into the program. You'll find them either in the properties for the PIDSubsystem in RobotBuilder or in the constructor for the PID Subsystem in your code.
+
+The F (feedforward) term is used for controlling velocity with a PID controller.
+
+You can find more information in Operating the robot with feedback from sensors(C++ or Java).
```

Comment:
```suggestion
.. todo:: add link to pid article when ported

You can find more information in Operating the robot with feedback from sensors.
```

Reply:
With this change, it's all good. The PID article has yet to be ported (likely to be rewritten).